src/Compiler/Driver/CompilerImports.fs

```diff
@@ -2229,21 +2229,25 @@ and [<Sealed>] TcImports
 
     // NOTE: When used in the Language Service this can cause the transitive checking of projects. Hence it must be cancellable.
     member tcImports.RegisterAndImportReferencedAssemblies(ctok, nms: AssemblyResolution list) =
-        node {
+        async {
             CheckDisposed()
 
+            // Prevent deadlocks in FSI.
```

Comment:
I'd love to put a more meaningful comment here, but I'm not 100% sure what's going on here and I don't want to misrepresent anything.

Reply:
I can't reproduce the problem that this was supposed to solve. I'll get rid of this and see what happens in CI.